workload/src/watcher.h

```diff
@@ -1,103 +1,104 @@
-#ifndef WATCHER_H
-#define WATCHER_H
+#ifndef INCLUDE_WORKLOAD_WATCHER_H
+#define INCLUDE_WORKLOAD_WATCHER_H
 
-#include "../../svid/x509svid/src/svid.h"
 #include "../../bundle/x509bundle/src/bundle.h"
 #include "../../bundle/x509bundle/src/set.h"
+#include "../../svid/x509svid/src/svid.h"
+#include "client.h"
+#include "x509context.h"
 
 #include "../../bundle/jwtbundle/src/bundle.h"
 #include "../../bundle/jwtbundle/src/set.h"
 
 #include <threads.h>
 #include <time.h>
 
-///TODO: function for picking first SVID
-typedef struct workloadapi_X509Context
-{
-    x509svid_SVID** SVIDs;
-    x509bundle_Set* Bundles;
-
-} workloadapi_X509Context;
+#ifdef __cplusplus
+extern "C" {
+#endif
 
-typedef struct workloadapi_WatcherConfig
-{
-    ///TODO: add actual client type
-    void* client;
-    ///TODO: add actual client option type
-    void** clientOptions;
+typedef struct workloadapi_Client workloadapi_Client;
+typedef void (*workloadapi_ClientOption)(workloadapi_Client *, void *);
+typedef struct workloadapi_WatcherConfig {
+    workloadapi_Client *client;
+    workloadapi_ClientOption *clientOptions;
 } workloadapi_WatcherConfig;
 
-// type for callback function. will be set by X509Source.
-typedef void (*workloadapi_x509ContextFunc_t)(workloadapi_X509Context*, void*); 
-// eg.: 
-// workloadapi_x509ContextFunc_t func; -> void (*func)(workloadapi_X509Context* updatedContext);
-
-typedef struct X509Callback{
-    void* args;
-    workloadapi_x509ContextFunc_t func;
-} workloadapi_X509Callback;
-
 // typedef void(workloadapi_jwtBundleSetFunc_t)(jwtbundle_Set*);
-// eg.: 
+// eg.:
 
-///TODO: define error codes.
+typedef struct workloadapi_Watcher {
+    // Workload API client
+    workloadapi_Client *client;
 
-typedef struct workloadapi_Watcher
-{
-    void* client; ///TODO: as above, pointer to Client C++ class;
-    bool ownsClient; //did this create its client?
+    // did this create its client?
+    bool ownsClient;
 
-    //Update Sync
+    // Update Sync
     mtx_t updateMutex;
     cnd_t updateCond;
     bool updated;
-    
-    ///TODO: needed?
-    err_t updateError;
 
+    err_t updateError;
 
-    //Close sync
+    // Close sync
     mtx_t closeMutex;
     bool closed;
-    err_t closeError; //needed?
-    
-    thrd_t watcherThread; //thread spun to wait on updates
+    err_t closeError;
+
+    // thread spun to wait on updates
+    thrd_t watcherThread;
     int threadError;
```

Comment:
Replace "closeError", "watcherThread" and "threadError" for "close_error", "watcher_thread" and "thread_error", respectively